enhancements/sha-bundle-digest.md

```diff
@@ -0,0 +1,136 @@
+---
+title: SHA-Bundle-Digest
+authors:
+  - '@theishshah'
+reviewers:
+  - '@camilamacedo86'
+  - '@joelanford'
+  - '@jmrodri'
+  - '@ryantking'
+  - '@tlwu2013'
+approvers:
+  - '@tlwu2013'
+  - '@jmrodri'
+  - '@joelanford'
+
+creation-date: 2022-01-14
+
+last-updated: 2022-01-14
+
+status: implementable
+---
+
+# Support SHA-based Image Tags
+
+## Summary
+
+Currently, operators generated by `operator-sdk` will store and retrieve container images based on the image name and
+the image tag in the bundle. This EP adds support to `operator-sdk` such that images can be stored and retrieved in the
+bundle with an image name and an image digest as an optional replacement for a tag. Image digests are SHA-256 sums that
+are hashed from the image layer therefore uniquely identifying one specific image.
+
+## Motivation
+
+Many users run Kubernetes in an air-gapped environment (also known as an offline or disconnected environment), which is
+when a cluster network does not have any access to the public internet, usually for security purposes. In this setup,
+the digest of an image will be different than it is in the public repository since the image location changes. For
+example, if a user imports the image `quay.io/golang:1.17` into their air-gapped registry as
+`quay.internaldomain.net/golang:1.17`, the digest of the image will be different due to the different name even though
+the contents of the images are identical. Using digest SHAs instead of tags in the air-gapped environment will keep the
+digest from changing registry to registry.
+
+### Goals
+
+- Give users the ability to enable digest-based image resolution
+- Allow users to specify additional images to resolve digests for.
+- Add ability to resolve image digests from image tag dynamically using local docker registry.
+
+### Non-Goals
+
+- Provide an automated way to migrate existing Operator-SDK-based operators from image tags to image digests.
+- Identify locations in user code that reference images by tag.
+
+## Proposal
+
+- Add `--use-image-digests` to `operator-sdk generate bundle` command. When enabled, the CLI will:
+  - Search the environment defined in `manager.yaml` for all variables that begin with `RELATED_IMAGE_` to build a list
+    of all images that need digests in the output cluster CSV.
+  - Resolve the digests for the operator image and any additional related images it finds in the environment.
+  - Output the images with digests instead of tags in the cluster CSV.
```

Comment:
```suggestion
  1. Search the environment defined in `manager.yaml` for all variables that begin with `RELATED_IMAGE_` to build a list
    of all related images that need to be added to the CSV (see Appendix C).
  2. Resolve the digests for the operator image and any additional related images it finds in the environment.
  3. Output the operator image(s) with digests instead of tags in the CSV.
  4. Add related images with their digests to the CSV.
```